tests/virtctl/credentials.go

```diff
@@ -0,0 +1,216 @@
+	)
+
+	const (
+		sshKeySecretName   = "test-secret-ssh"
+		passwordSecretName = "test-secret-password"
+		userName           = "root"
+	)
+
+	var (
+		cli kubecli.KubevirtClient
+
+		vm *kubevirtv1.VirtualMachine
+	)
+
+	BeforeEach(func() {
+		cli = kubevirt.Client()
+
+		keySecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: sshKeySecretName,
+			},
+			Data: map[string][]byte{
+				"key-file.pub": []byte(testKey1),
+			},
+		}
+
+		_, err := cli.CoreV1().Secrets(util.NamespaceTestDefault).Create(context.Background(), keySecret, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		passwordSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: passwordSecretName,
+			},
+			Data: map[string][]byte{
+				userName: []byte("test-password"),
+			},
+		}
+
+		_, err = cli.CoreV1().Secrets(util.NamespaceTestDefault).Create(context.Background(), passwordSecret, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		vmi := libvmi.NewFedora()
+		vm = tests.NewRandomVirtualMachine(vmi, false)
+
+		vm.Spec.Template.Spec.AccessCredentials = []kubevirtv1.AccessCredential{{
+			SSHPublicKey: &kubevirtv1.SSHPublicKeyAccessCredential{
+				Source: kubevirtv1.SSHPublicKeyAccessCredentialSource{
+					Secret: &kubevirtv1.AccessCredentialSecretSource{
+						SecretName: sshKeySecretName,
+					},
+				},
+				PropagationMethod: kubevirtv1.SSHPublicKeyAccessCredentialPropagationMethod{
+					QemuGuestAgent: &kubevirtv1.QemuGuestAgentSSHPublicKeyAccessCredentialPropagation{
+						Users: []string{userName},
+					},
+				},
+			},
+		}, {
+			UserPassword: &kubevirtv1.UserPasswordAccessCredential{
+				Source: kubevirtv1.UserPasswordAccessCredentialSource{
+					Secret: &kubevirtv1.AccessCredentialSecretSource{
+						SecretName: passwordSecretName,
+					},
+				},
+				PropagationMethod: kubevirtv1.UserPasswordAccessCredentialPropagationMethod{
+					QemuGuestAgent: &kubevirtv1.QemuGuestAgentUserPasswordAccessCredentialPropagation{},
+				},
+			},
+		}}
+
+		vm, err = cli.VirtualMachine(util.NamespaceTestDefault).Create(context.Background(), vm)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("add-ssh-key", func() {
+		It("[test_id:TODO] should add ssh key to a secret", func() {
+			err := clientcmd.NewRepeatableVirtctlCommand(
+				"credentials", "add-ssh-key",
+				"--user", userName,
+				"--value", testKey2,
+				vm.Name,
+			)()
+			Expect(err).ToNot(HaveOccurred())
+
+			secret, err := cli.CoreV1().Secrets(util.NamespaceTestDefault).Get(context.Background(), sshKeySecretName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(secret.Data).To(HaveLen(2))
+			Expect(secret.Data).To(HaveValue(Equal([]byte(testKey2))))
+		})
+
+		It("[test_id:TODO] should add ssh key from a file", func() {
+			var filename string
+			func() {
+				file, err := os.CreateTemp("", "test-key-file-")
```

Comment:
You have a similar function for adding and removing the key. Additionally, you could use the `TempDir` function from ginkgo, in this way, it gets automatically cleaned up after the test. Like in https://github.com/kubevirt/kubevirt/blob/main/tests/virtctl/ssh.go#L74